contracts/interchain-token-service/tests/deploy_remote_canonical_token.rs

```diff
@@ -0,0 +1,89 @@
+mod utils;
+
+use axelar_soroban_std::{address::AddressExt, events};
+use interchain_token_service::{
+    event::InterchainTokenDeploymentStartedEvent, types::TokenManagerType,
+};
+use soroban_sdk::{testutils::Address as _, Address, BytesN, String};
+use soroban_token_sdk::metadata::TokenMetadata;
+use utils::{setup_env, setup_gas_token, TokenMetadataExt};
+
+#[test]
+fn deploy_remote_canonical_token_succeeds() {
+    let (env, client, _, _, _) = setup_env();
+    env.mock_all_auths();
```

Comment:
avoid mock_all_auths, see recent tests that were added, we should mock when necessary, and for successful execution test, auth should be tested